src/gluonnlp/vocab/bert.py

```diff
@@ -97,113 +110,111 @@ class BERTVocab(Vocab):
+        if mask_token:
+            assert sep_token not in identifiers_to_tokens
+            identifiers_to_tokens.update(sep_token=sep_token)
+        if mask_token:
+            assert cls_token not in identifiers_to_tokens
+            identifiers_to_tokens.update(cls_token=cls_token)
+
+        super(BERTVocab, self).__init__(counter=counter, max_size=max_size,
+                                        min_freq=min_freq, unknown_token=unknown_token,
+                                        padding_token=padding_token, bos_token=bos_token,
+                                        eos_token=eos_token,
+                                        reserved_tokens=reserved_tokens,
+                                        identifiers_to_tokens=identifiers_to_tokens,
+                                        token_to_idx=token_to_idx)
 
     @classmethod
     def from_json(cls, json_str):
-        """Deserialize BERTVocab object from json string.
+        """Deserialize Vocab object from json string.
 
         Parameters
         ----------
         json_str : str
-            Serialized json string of a BERTVocab object.
+            Serialized json string of a Vocab object.
+
 
         Returns
         -------
-        BERTVocab
+        Vocab
         """
         vocab_dict = json.loads(json_str)
-
+        token_to_idx = vocab_dict.get('token_to_idx')
         unknown_token = vocab_dict.get('unknown_token')
-        bert_vocab = cls(unknown_token=unknown_token)
-        bert_vocab._idx_to_token = vocab_dict.get('idx_to_token')
-        bert_vocab._token_to_idx = vocab_dict.get('token_to_idx')
-        if unknown_token:
-            bert_vocab._token_to_idx = DefaultLookupDict(bert_vocab._token_to_idx[unknown_token],
-                                                         bert_vocab._token_to_idx)
-        bert_vocab._reserved_tokens = vocab_dict.get('reserved_tokens')
-        bert_vocab._padding_token = vocab_dict.get('padding_token')
-        bert_vocab._bos_token = vocab_dict.get('bos_token')
-        bert_vocab._eos_token = vocab_dict.get('eos_token')
-        bert_vocab._mask_token = vocab_dict.get('mask_token')
-        bert_vocab._sep_token = vocab_dict.get('sep_token')
-        bert_vocab._cls_token = vocab_dict.get('cls_token')
-
-        return bert_vocab
+        padding_token = vocab_dict.get('padding_token')
+        bos_token = vocab_dict.get('bos_token')
+        eos_token = vocab_dict.get('eos_token')
+        special_tokens = [unknown_token, padding_token, bos_token, eos_token]
+        reserved_tokens = vocab_dict.get('reserved_tokens')
+
+        # workaround reserved and special tokens being serialized together
+        reserved_tokens = [
+            t for t in reserved_tokens if t not in special_tokens
+        ]
+
+        # Backwards compatibility with BERTVocab serialization in GluonNLP 0.7
+        identifiers_to_tokens = vocab_dict.get('identifiers_to_tokens')
```

Comment:
This does not exist in previous vocab files

Reply:
Do you mean the code did not exist? It is to enable loading the `BERTVocab` files that were created with GluonNLP 0.7; for example the ones distributed via S3. Or do you mean the 3 special tokens below did not exist in the serialized files?

Reply:
I meant GluonNLP 0.6

Reply:
Oh I meant the `identifiers_to_tokens` field did not exist. can we provide a default value?